tests/addr_parse.rs

```diff
@@ -0,0 +1,97 @@
+use std::collections::HashSet;
+use std::iter::FromIterator;
+
+use mdns_sd::AsIpv4Addrs;
+use nix::sys::socket::Ipv4Addr;
+
+#[test]
+fn test_addr_str() {
+    assert_eq!(
+        "127.0.0.1".as_ipv4_addrs(),
+        Ok({
+            let mut set = HashSet::new();
+            set.insert(Ipv4Addr::from_std(&std::net::Ipv4Addr::new(127, 0, 0, 1)));
+
+            set
+        })
+    );
+
+    let addr = "127.0.0.1".to_string();
+    assert_eq!(
+        addr.as_ipv4_addrs(),
+        Ok(HashSet::from_iter([Ipv4Addr::from_std(
```

Comment:
Until Rust 1.53, only references to arrays implement IntoIterator ([doc](https://doc.rust-lang.org/edition-guide/rust-2021/IntoIterator-for-arrays.html)). You should be able to use `(&[array]).into_iter()` .

Reply:
o yea, I fixed it in the other examples via an explicit construction of the `HashSet`, but I must've overlooked that one